xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp

```diff
@@ -236,15 +236,26 @@ bool CDVDVideoCodecFFmpeg::Open(CDVDStreamInfo &hints, CDVDCodecOptions &options
   // setup threading model
   if (!hints.software)
   {
-    if ((EDECODEMETHOD)CSettings::Get().GetInt("videoplayer.decodingmethod") == VS_DECODEMETHOD_HARDWARE &&
-        m_decoderState == STATE_NONE)
+    bool tryhw = false;
+#ifdef HAVE_LIBVDPAU
+    if(CSettings::Get().GetBool("videoplayer.usevdpau"))
+      tryhw = true;
+#endif
+#ifdef HAVE_LIBVA
+    if(CSettings::Get().GetBool("videoplayer.usevaapi"))
+      tryhw = true;
+#endif
+#ifdef HAS_DX
+    if(CSettings::Get().GetBool("videoplayer.usedxva2"))
+      tryhw = true;
+#endif
+#ifdef TARGET_DARWIN_OSX
+    if(CSettings::Get().GetBool("videoplayer.usevda"))
+      tryhw = true;
+#endif
+    if (tryhw && m_decoderState == STATE_NONE)
     {
-#if defined(TARGET_ANDROID) || defined(TARGET_DARWIN_IOS)
-      // If we get here on Android or iOS, it's always multi
-      m_decoderState = STATE_SW_MULTI;
```

Comment:
it is not really dropped, the logic changed. tryhw is initialised to false and not set to true for those platforms.